vault-core/src/main/java/org/apache/jackrabbit/vault/packaging/registry/impl/FSInstallStateCache.java

```diff
@@ -55,8 +60,13 @@
     private final Path homeDir;
 
     public FSInstallStateCache(Path homeDir) throws IOException {
-        this.homeDir = homeDir;
-        Files.createDirectories(homeDir.toRealPath());
+        try {
+            this.homeDir = homeDir;
+            Files.createDirectories(homeDir.toRealPath());
+        } catch (FileAlreadyExistsException ex) {
+            log.error("on createDirectories for " + homeDir + " (real path: " + homeDir.toRealPath() + ")", ex);
```

Comment:
Let us rather wrap the original exception and add this detail message. Both throwing and logging is an antipattern and leads to redundant log entries.

Reply:
Any preference about that Exception to wrap into?

Reply:
A generic IOException would be fine for me.